packages/electron-renderer/src/use-node.js.ts

```diff
@@ -88,31 +88,57 @@ const electron = `
  * All exports module see https://www.electronjs.org -> API -> Renderer Process Modules
  */
 const electron = require("electron");
-const {
-  clipboard,
-  nativeImage,
-  shell,
-  contextBridge,
-  crashReporter,
-  ipcRenderer,
-  webFrame,
-  desktopCapturer,
-  deprecate,
-} = electron;
 
-export {
-  electron as default,
-  clipboard,
-  nativeImage,
-  shell,
-  contextBridge,
-  crashReporter,
-  ipcRenderer,
-  webFrame,
-  desktopCapturer,
-  deprecate,
+// Proxy in Worker
+let _ipcRenderer;
+if (typeof document === 'undefined') {
+  _ipcRenderer = {};
+  const keys = [
+    'invoke',
+    'postMessage',
+    'send',
+    'sendSync',
+    'sendTo',
+    'sendToHost',
+    // propertype
+    'addListener',
+    'emit',
+    'eventNames',
+    'getMaxListeners',
+    'listenerCount',
+    'listeners',
+    'off',
+    'on',
+    'once',
+    'prependListener',
+    'prependOnceListener',
+    'rawListeners',
+    'removeAllListeners',
+    'removeListener',
+    'setMaxListeners',
+  ];
+  for (const key of keys) {
+    _ipcRenderer[key] = () => {
+      throw new Error(
+        '"ipcRenderer" does not work in "Worker"\\n' +
+        'You can see https://github.com/electron-vite/vite-plugin-electron/issues/69'
+      );
```

Comment:
@astoilkov 👋
Not sure if the throw message is accurate. 🤔

Reply:
I would update the first sentence like that: "ipcRenderer does not work in a Web Worker". The second one is good.

Reply:
Can you a submit PR to v0.9.3 branch 😅